src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IAsyncEnumerableConverterFactory.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization.Converters;
+
+namespace System.Text.Json.Serialization
+{
+    /// <summary>
+    /// Converter for streaming <see cref="IAsyncEnumerable{T}" /> values.
+    /// </summary>
+    internal sealed class IAsyncEnumerableConverterFactory : JsonConverterFactory
+    {
+        public override bool CanConvert(Type typeToConvert) => TryGetAsyncEnumerableInterface(typeToConvert, out _);
+
+        public override JsonConverter CreateConverter(Type typeToConvert, JsonSerializerOptions options)
+        {
+            TryGetAsyncEnumerableInterface(typeToConvert, out Type? asyncEnumerableInterface);
+            Debug.Assert(asyncEnumerableInterface is not null, $"{typeToConvert} not supported by converter.");
+
+            Type elementType = asyncEnumerableInterface.GetGenericArguments()[0];
+            Type converterType = typeof(IAsyncEnumerableOfTConverter<,>).MakeGenericType(typeToConvert, elementType);
+            return (JsonConverter)Activator.CreateInstance(converterType)!;
+        }
+
+        internal static bool TryGetAsyncEnumerableInterface(Type type, [NotNullWhen(true)] out Type? asyncEnumerableInterface)
```

Comment:
Does `IEnumerableConverterFactoryHelpers.GetCompatibleGenericInterface` perform the same function as this method?

Reply:
Looks like it does. I'll change it to that.